addons/website/static/src/snippets/s_embed_code/000.js

```diff
@@ -0,0 +1,41 @@
+/** @odoo-module **/
+
+import publicWidget from "web.public.widget";
+import { _t } from "@web/core/l10n/translation";
+
+const EmbedCodeWidget = publicWidget.Widget.extend({
+    selector: ".s_embed_code",
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    async start() {
+        this.embedCodeEl = this.el.querySelector(".s_embed_code_embedded");
+        if (this.editableMode && this.embedCodeEl.offsetHeight === 0) {
+            this._addPlaceholder();
+        }
+        return await this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        const embedContent = this.embedCodeEl.dataset.embedContent;
+        if (!this.editableMode && embedContent && embedContent.match(/<script[^<]*?>/)) {
+            this.embedCodeEl.innerHTML = embedContent.replace(/<script[^<]*?>.*?<\/script>/gs, "");
```

Comment:
Actually maybe the easiest and more robust would be to parse the `dataset.embedContent` into a JS node (whose script won't be executed while not in the DOM) and use `querySelector` to search for the script? Especially since you do more manipulation on this line?

Reply:
Turns out that the DOMParser automatically strips off the `script` tags, so it's even simpler.